src/components/shared/Popover.js

```diff
@@ -15,16 +17,18 @@ class Popover extends Component {
 
   componentDidMount() {
     const { type } = this.props;
-    const { left, top, dir } = type == "popover"
+    const { left, top, dir, targetMid } = type == "popover"
       ? this.getPopoverCoords()
       : this.getTooltipCoords();
 
     // eslint-disable-next-line react/no-did-mount-set-state
-    this.setState({ left, top, dir });
+    this.setState({ left, top, dir, targetMid });
   }
 
   getPopoverCoords() {
     const el = ReactDOM.findDOMNode(this);
+    const wrapper = document.getElementsByClassName("editor-wrapper")[0];
```

Comment:
good intuition! I agree, using the document API feels weird for this. Let me investigate a bit and I will get back to you